### DISCUSSION TITLE
Custom Utilities and dbt Transforms Not Being Tracked in Meltano State

### DISCUSSION THREAD

**Discussion by mistermax80** [meltano/meltano#8984] [Q&A]
Hi Meltano team,

I’ve noticed that neither dbt transforms nor custom utilities (like calculate_free_slot:run_script) are being tracked in Meltano’s state database when included in a job.

For example, I have a job configured like this:

yaml
Copia codice
jobs:
- name: complete_elt
  tasks:
  - tap-mysql target-postgres dbt-postgres:run calculate_free_slot:run_script
The job runs successfully, but only the tap-to-target step gets recorded in the Meltano state database. Both the dbt-postgres:run and the custom utility (calculate_free_slot:run_script) do not appear in the state or as tracked runs in the database.

Is this expected behavior? If not, how can I ensure that these steps are tracked properly in Meltano's state database?

Thanks for your help!

Best regards,
Massimo

**Comment by tayloramurphy** [comment]
@mistermax80 the state db is only used for tracking incremental loading of data movement jobs. What sort of data are you wanting to track for utilities? cc @edgarrmondragon in case this was answered on Slack and can be closed.

**Reply by mistermax80** [reply]
> @mistermax80 the state db is only used for tracking incremental loading of data movement jobs. What sort of data are you wanting to track for utilities? cc @edgarrmondragon in case this was answered on Slack and can be closed.

I would like track dbt and custom python plugin.

**Reply by edgarrmondragon** [reply]
@mistermax80 I'm curious what is the to track from a dbt run. The tap-to-target parts of a job are tracked in the state to allow [incremental replication](https://docs.meltano.com/guide/integration/#incremental-replication-state), but dbt runs are usually stateless.

Regarding the custom utility plugin, I'm curious what state about it you'd like to track.

Related issues:

- https://github.com/meltano/meltano/issues/3439

**Reply by mistermax80** [reply]
So there is a misunderstanding... I apologise for this.
I mean the execution status, in runs table...

**Reply by edgarrmondragon** [reply]
No worries!

So, that table looks like this for the EL part:

```console
$ sqlite3 -markdown .meltano/meltano.db 'select id, job_name, state from runs order by started_at desc;'
| id |             job_name              |  state  |
|----|-----------------------------------|---------|
| 1  | dev:tap-getpocket-to-target-jsonl | SUCCESS |
```

Are you mostly interested in recording the job name (e.g. `dev:dbt`) and the state?

**Comment by mistermax80** [comment]
yes indeed I would like (even on different rows, but same run-id) to generate the log for each block in the run pipeline (dbt result, e custom python script result).
That table is very useful to monitor the executions and can be shown to the end user, but now it is "incomplete".
For the moment I am inserting custom code to write in the run table, but the run-id is regenerated, it would be nice to have it all managed by the "run" command.
